lineapy/db/db.py

```diff
@@ -602,3 +620,67 @@ def get_source_code_for_session(self, session_id: LineaID) -> str:
                 if script_source_code_orms is not None
                 else ""
             )
+
+    def delete_artifact_by_name(
+        self, artifact_name: str, version: Union[int, str] = None
+    ):
+        """
+        Deletes the most recent artifact with a certain name.
```

Comment:
The description doesn't include a reference to `"all"` and `"latest"`?

I don't really like the design using strings but it's fine for now.